src/libPMacc/include/cuSTL/container/allocator/DeviceMemAllocator.tpp

```diff
@@ -45,7 +45,8 @@ DeviceMemAllocator<Type, T_dim>::allocate(const math::Size_t<T_dim>& size)
     {
         cudaData.xsize = size[0];
         cudaData.ysize = size[1];
-        CUDA_CHECK_NO_EXCEP(cudaMallocPitch(&cudaData.ptr, &cudaData.pitch, cudaData.xsize * sizeof (Type), cudaData.ysize));
+        if(size.productOfComponents())
+            CUDA_CHECK_NO_EXCEP(cudaMallocPitch(&cudaData.ptr, &cudaData.pitch, cudaData.xsize * sizeof (Type), cudaData.ysize));
```

Comment:
Why did you use `CUDA_CHECK_NO_EXCEP` instead to throw an exception on error?
Is there a reason to ignore the error?